Makefile

```diff
@@ -19,7 +21,26 @@ machine-controller: $(shell find cmd pkg -name '*.go') vendor
 				github.com/kubermatic/machine-controller/cmd/controller
 
 docker-image: machine-controller
+	make docker-image-nodep
+
+docker-image-nodep:
```

Comment:
what means nodep?

Reply:
Yeah, thats rather non-obvious. The reason is that make uses timestamps on files to determine if it has to rebuild something. CircleCI uses caches from any build, which means that sometimes it restores the `vendor` folder from an older build. The result is:

* Make sees `Gopkg.{toml,lock}` are newere than the vender folder
* -> Have to build the vendor target
* -> Vendor target is newer than the binary ->Have to rebuild the binary

To skip all that (which makes perfect sense locally btw), I added that `-nodep` target